src/App.tsx

```diff
@@ -1,61 +1,148 @@
+import React, { useState } from 'react';
+import { TodoList } from './components/TodoList/TodoList';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { Todo } from './types/Todo';
+import { FullTodo } from './types/FullTodo';
+
+function getTodoList(todos: Todo[]):(FullTodo)[] {
+  return todos.map(todo => {
+    return {
+      ...todo,
+      user: usersFromServer.find(user => user.id === todo.userId) || null,
+    };
+  });
+}
+
+function getNextTodoId(todos: FullTodo[]): number {
+  if (todos.length === 0) {
+    return 1;
+  }
+
+  const maxId = Math.max(...todos.map(todo => todo.id));
+
+  return maxId + 1;
+}
+
+export const App: React.FC = () => {
+  const [formTitle, setFormTitle] = useState('');
+  const [todosList, setTodosList] = useState<Todo[]>([...todosFromServer]);
+  const [userId, setUserId] = useState(0);
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [hasSelectError, setHasSelectError] = useState(false);
+  const [count, setCount] = useState(0);
+
+  const todos = getTodoList(todosList);
+
+  const handleFormSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    let titleError = false;
+    let selectError = false;
+
+    if (formTitle === '') {
+      titleError = true;
+    }
+
+    if (userId === 0) {
+      selectError = true;
+    }
+
+    setHasTitleError(titleError);
+    setHasSelectError(selectError);
+
+    if (!titleError && !selectError) {
+      const newTodo = {
+        id: getNextTodoId(todos),
+        title: formTitle,
+        userId,
+        completed: false,
+      };
+
+      setTodosList((prevTodos) => [...prevTodos, newTodo]);
+
+      setFormTitle('');
+      setUserId(0);
+    }
+
+    setCount(count + 1);
```

Comment:
As I understand we don't use count as a key prop on form

Reply:
Maybe, just did it like previous task)